### PR TITLE
refactor: remove lerna and unify packages (LF-4923)

### PR DESCRIPTION
**Pull request by dennyscode**
Ticket: https://lifi.atlassian.net/browse/LF-4923

### CODE REVIEW

**Comment by dennyscode** [comment]
@dnnnvx With this ticket, we want to get rid of our lerna setup and move it all to the root folder. This obviously messes up the build process. Prepare, you will be needed :D

image

**Comment by dnnnvx** [comment]
@dennyscode I pushed in this branch the actions fixes 😁

**Comment by dennyscode** [comment]
- [ ] husky working

**Comment by Addminus** [comment]
@dennyscode Any info on what is still needed to be done here?

**Comment by dennyscode** [comment]
Still getting bunch of errors on lint-staged script when committing. Build runs fine

**Comment by dennyscode** [comment]
Still running into this issue when lint-staged runs:

```
> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file -
Executing pre-commit scripts
[33m[STARTED][39m Preparing lint-staged...
[32m[COMPLETED][39m Preparing lint-staged...
[33m[STARTED][39m Running tasks for staged files...
[33m[STARTED][39m package.json — 16 files
[33m[STARTED][39m **/*.(ts|tsx|js) — 12 files
[33m[STARTED][39m **/*.(md|json) — 3 files
[33m[STARTED][39m yarn tsc --noEmit
[33m[STARTED][39m yarn prettier --write
[32m[COMPLETED][39m yarn prettier --write
[32m[COMPLETED][39m **/*.(md|json) — 3 files
[31m[FAILED][39m yarn tsc --noEmit [FAILED]
[31m[FAILED][39m yarn tsc --noEmit [FAILED]
[32m[COMPLETED][39m Running tasks for staged files...
[33m[STARTED][39m Applying modifications from tasks...
[33m[SKIPPED][39m Skipped because of errors from tasks.
[33m[STARTED][39m Reverting to original state because of errors...
[32m[COMPLETED][39m Reverting to original state because of errors...
[33m[STARTED][39m Cleaning up temporary files...
[32m[COMPLETED][39m Cleaning up temporary files...

× yarn tsc --noEmit:
node_modules/@lifi/widget/i18n/index.d.ts(1,16): error TS2732: Cannot find module './bn.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(2,16): error TS2732: Cannot find module './de.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(3,16): error TS2732: Cannot find module './en.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(4,16): error TS2732: Cannot find module './es.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(5,16): error TS2732: Cannot find module './fr.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(6,16): error TS2732: Cannot find module './id.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(7,16): error TS2732: Cannot find module './it.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(8,16): error TS2732: Cannot find module './ko.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(9,16): error TS2732: Cannot find module './pt.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(10,16): error TS2732: Cannot find module './th.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(11,16): error TS2732: Cannot find module './tr.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(12,16): error TS2732: Cannot find module './uk.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(13,16): error TS2732: Cannot find module './vi.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/i18n/index.d.ts(14,16): error TS2732: Cannot find module './zh.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
node_modules/@lifi/widget/types/widget.d.ts(3,56): error TS2305: Module '"@mui/material"' has no exported member 'Shape'.
node_modules/@mui/utils/chainPropTypes/chainPropTypes.d.ts(1,8): error TS1192: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/prop-types/index"' has no default export.
node_modules/@mui/utils/elementAcceptingRef.d.ts(1,8): error TS1192: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/prop-types/index"' has no default export.
node_modules/@mui/utils/elementTypeAcceptingRef.d.ts(1,8): error TS1192: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/prop-types/index"' has no default export.
node_modules/@mui/utils/integerPropType.d.ts(1,8): error TS1192: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/prop-types/index"' has no default export.
node_modules/@mui/utils/refType.d.ts(1,8): error TS1192: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/prop-types/index"' has no default export.
node_modules/@walletconnect/modal/dist/_types/src/client.d.ts(12,34): error TS2307: Cannot find module 'packages/modal-core/dist/_types/src/controllers/ModalCtrl' or its corresponding type declarations.
node_modules/@walletconnect/modal/dist/_types/src/client.d.ts(14,50): error TS2307: Cannot find module 'packages/modal-core/dist/_types/src/types/controllerTypes' or its corresponding type declarations.
node_modules/@walletconnect/types/dist/types/core/pairing.d.ts(3,8): error TS1259: Module '"events"' can only be default-imported using the 'esModuleInterop' flag
node_modules/@walletconnect/types/dist/types/sign-client/client.d.ts(3,8): error TS1259: Module '"events"' can only be default-imported using the 'esModuleInterop' flag
node_modules/@walletconnect/universal-provider/dist/types/UniversalProvider.d.ts(6,8): error TS1259: Module '"events"' can only be default-imported using the 'esModuleInterop' flag
node_modules/react-hook-form/dist/form.d.ts(1,8): error TS1259: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/react/index"' can only be default-imported using the 'esModuleInterop' flag
node_modules/react-hook-form/dist/types/controller.d.ts(1,8): error TS1259: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/react/index"' can only be default-imported using the 'esModuleInterop' flag
node_modules/react-hook-form/dist/types/form.d.ts(1,8): error TS1259: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/react/index"' can only be default-imported using the 'esModuleInterop' flag
node_modules/react-hook-form/dist/useFormContext.d.ts(1,8): error TS1259: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/react/index"' can only be default-imported using the 'esModuleInterop' flag
src/config/config.ts(2,55): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
src/const/MenuConfig/useWalletSelectContent.tsx(6,29): error TS2307: Cannot find module 'src/hooks' or its corresponding type declarations.
src/const/MenuConfig/useWalletSelectContent.tsx(7,27): error TS2307: Cannot find module 'src/providers' or its corresponding type declarations.
src/const/MenuConfig/useWalletSelectContent.tsx(8,48): error TS2307: Cannot find module 'src/stores' or its corresponding type declarations.
src/const/MenuConfig/useWalletSelectContent.tsx(9,35): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
src/const/MenuConfig/useWalletSelectContent.tsx(103,11): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/hooks/useChains.ts(14,22): error TS1343: The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'es2022', 'esnext', 'system', 'node16', or 'nodenext'.
src/hooks/useChains.ts(14,34): error TS2339: Property 'env' does not exist on type 'ImportMeta'.
src/i18n/index.ts(1,16): error TS2732: Cannot find module './bn/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(1,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(2,16): error TS2732: Cannot find module './en/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(2,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(3,16): error TS2732: Cannot find module './es/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(3,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(4,16): error TS2732: Cannot find module './fr/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(4,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(5,16): error TS2732: Cannot find module './id/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(5,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(6,16): error TS2732: Cannot find module './it/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(6,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(7,16): error TS2732: Cannot find module './ko/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(7,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(8,16): error TS2732: Cannot find module './pt/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(8,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(9,16): error TS2732: Cannot find module './th/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(9,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(10,16): error TS2732: Cannot find module './tr/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(10,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(11,16): error TS2732: Cannot find module './uk/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(11,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(12,16): error TS2732: Cannot find module './vi/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(12,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/i18n/index.ts(13,16): error TS2732: Cannot find module './zh/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(13,37): error TS2821: Import assertions are only supported when the '--module' option is set to 'esnext' or 'nodenext'.
src/index.tsx(4,28): error TS2307: Cannot find module 'src/config' or its corresponding type declarations.
src/index.tsx(7,21): error TS6142: Module './App' was resolved to '[...]/Li.Fi/jumper.exchange/src/App.tsx', but '--jsx' is not set.
src/index.tsx(17,3): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/index.tsx(18,5): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/index.tsx(25,5): error TS1343: The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'es2022', 'esnext', 'system', 'node16', or 'nodenext'.
src/index.tsx(25,17): error TS2339: Property 'env' does not exist on type 'ImportMeta'.
src/organisms/FeatureCards/FeatureCards.tsx(4,29): error TS2307: Cannot find module 'src/components' or its corresponding type declarations.
src/organisms/FeatureCards/FeatureCards.tsx(5,33): error TS2307: Cannot find module 'src/hooks' or its corresponding type declarations.
src/organisms/FeatureCards/FeatureCards.tsx(6,34): error TS2307: Cannot find module 'src/stores' or its corresponding type declarations.
src/organisms/FeatureCards/FeatureCards.tsx(7,38): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
src/organisms/FeatureCards/FeatureCards.tsx(9,10): error TS2305: Module '"."' has no exported member 'FeatureCardsContainer'.
src/organisms/FeatureCards/FeatureCards.tsx(44,7): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/FeatureCards/FeatureCards.tsx(47,13): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/FeatureCards/index.ts(1,15): error TS6142: Module './FeatureCards' was resolved to '[...]/Li.Fi/jumper.exchange/src/organisms/FeatureCards/FeatureCards.tsx', but '--jsx' is not set.
src/organisms/FeatureCards/index.ts(2,15): error TS6142: Module './FeatureCards.style' was resolved to '[...]/Li.Fi/jumper.exchange/src/organisms/FeatureCards/FeatureCards.style.tsx', but '--jsx' is not set.
src/organisms/Header/Header.tsx(2,52): error TS2307: Cannot find module 'src/components' or its corresponding type declarations.
src/organisms/Header/Header.tsx(3,27): error TS2307: Cannot find module 'src/providers' or its corresponding type declarations.
src/organisms/Header/Header.tsx(4,34): error TS2307: Cannot find module 'src/stores' or its corresponding type declarations.
src/organisms/Header/Header.tsx(5,62): error TS6142: Module '.' was resolved to '[...]/Li.Fi/jumper.exchange/src/organisms/Header/index.tsx', but '--jsx' is not set.
src/organisms/Header/Header.tsx(19,5): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Header/Header.tsx(20,7): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Header/Header.tsx(21,9): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Header/Header.tsx(23,7): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Header/Header.tsx(24,7): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(5,50): error TS2307: Cannot find module 'src/components' or its corresponding type declarations.
src/organisms/Widgets/Widgets.tsx(6,59): error TS2307: Cannot find module 'src/const' or its corresponding type declarations.
src/organisms/Widgets/Widgets.tsx(7,33): error TS2307: Cannot find module 'src/hooks' or its corresponding type declarations.
src/organisms/Widgets/Widgets.tsx(8,61): error TS2307: Cannot find module 'src/organisms' or its corresponding type declarations.
src/organisms/Widgets/Widgets.tsx(9,53): error TS2307: Cannot find module 'src/stores' or its corresponding type declarations.
src/organisms/Widgets/Widgets.tsx(10,41): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
src/organisms/Widgets/Widgets.tsx(11,44): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
src/organisms/Widgets/Widgets.tsx(101,5): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(105,8): error TS1343: The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'es2022', 'esnext', 'system', 'node16', or 'nodenext'.
src/organisms/Widgets/Widgets.tsx(105,20): error TS2339: Property 'env' does not exist on type 'ImportMeta'.
src/organisms/Widgets/Widgets.tsx(106,9): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(107,11): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(110,7): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(115,9): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(117,7): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(122,9): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(124,8): error TS1343: The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'es2022', 'esnext', 'system', 'node16', or 'nodenext'.
src/organisms/Widgets/Widgets.tsx(124,20): error TS2339: Property 'env' does not exist on type 'ImportMeta'.
src/organisms/Widgets/Widgets.tsx(125,9): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(131,11): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(132,13): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/organisms/Widgets/Widgets.tsx(136,7): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
src/stores/settings/SettingsStore.tsx(2,46): error TS2307: Cannot find module 'src/config' or its corresponding type declarations.
src/stores/settings/SettingsStore.tsx(9,8): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
src/stores/settings/SettingsStore.tsx(31,24): error TS2550: Property 'hasOwn' does not exist on type 'ObjectConstructor'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2022' or later.
src/types/settings.ts(3,34): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
src/utils/sortByName.ts(1,31): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
src/utils/structuredClone.ts(2,8): error TS1259: Module '"[...]/Li.Fi/jumper.exchange/node_modules/@types/ungap__structured-clone/index"' can only be default-imported using the 'esModuleInterop' flag
src/utils/walletDigest.ts(1,36): error TS2307: Cannot find module 'src/types' or its corresponding type declarations.
husky - pre-commit hook exited with code 1 (error)
```

File: src/i18n/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import fr from './fr/language.json' assert { type: 'json' };
+import id from './id/language.json' assert { type: 'json' };
+import it from './it/language.json' assert { type: 'json' };
+import ko from './ko/language.json' assert { type: 'json' };
+import pt from './pt/language.json' assert { type: 'json' };
+import th from './th/language.json' assert { type: 'json' };
+import tr from './tr/language.json' assert { type: 'json' };
+import uk from './uk/language.json' assert { type: 'json' };
+import vi from './vi/language.json' assert { type: 'json' };
+import zh from './zh/language.json' assert { type: 'json' };
```
Comment (dennyscode) [inline src/i18n/index.ts:13]: @chybisov Does it make sense to fix it that way, using that assert type json:

```
src/i18n/index.ts(1,16): error TS2732: Cannot find module './bn/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(2,16): error TS2732: Cannot find module './en/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(3,16): error TS2732: Cannot find module './es/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(4,16): error TS2732: Cannot find module './fr/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(5,16): error TS2732: Cannot find module './id/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(6,16): error TS2732: Cannot find module './it/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(7,16): error TS2732: Cannot find module './ko/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(8,16): error TS2732: Cannot find module './pt/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(9,16): error TS2732: Cannot find module './th/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(10,16): error TS2732: Cannot find module './tr/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(11,16): error TS2732: Cannot find module './uk/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(12,16): error TS2732: Cannot find module './vi/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
src/i18n/index.ts(13,16): error TS2732: Cannot find module './zh/language.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.
```

Comment (chybisov) [reply src/i18n/index.ts:13]: @dennyscode should we use `"resolveJsonModule": true` is tsconfig as suggested in error?

**Review by Addminus — APPROVED** [review]

**Review by chybisov — APPROVED** [review]

**Pull request merged**